rotkehlchen/chain/ethereum/modules/curve/curve_cache.py

```diff
@@ -215,172 +254,190 @@ def save_curve_data_to_cache(
-    # so no need to query underlying tokens
-    pool_coins_result = ethereum.multicall_specific(
-        contract=factory_contract,
-        method_name='get_coins',
-        arguments=[(x,) for x in pool_addresses],
-    )
-    pools_mapping: CURVE_POOLS_MAPPING_TYPE = {}
-    for pool_addr, (decoded_coins_result,) in zip(pool_addresses, pool_coins_result):
-        if pool_addr in IGNORED_CURVE_POOLS:
+    pool_count = metaregistry.call(node_inquirer=ethereum, method_name='pool_count')
+    all_pools = []
+    for pool_index in range(pool_count):
+        pool_address = to_checksum_address(metaregistry.call(
+            node_inquirer=ethereum,
+            method_name='pool_list',
+            arguments=[pool_index],
+        ))
+        if pool_address in IGNORED_CURVE_POOLS:
             continue
-        pool_coins = []
-        for coin_addr in decoded_coins_result:
-            checksumed_coin_addr = to_checksum_address(coin_addr)
-            if checksumed_coin_addr == ZERO_ADDRESS:
-                # curve returns array of fixed length. So if current pool utilizes fewer
-                # coins than the array's length, first values in the result are the real
-                # coins and all addresses after are just 0x00..00
+        pool_name = metaregistry.call(
+            node_inquirer=ethereum,
+            method_name='get_pool_name',
+            arguments=[pool_address],
+        )
+        gauge_address = to_checksum_address(metaregistry.call(
+            node_inquirer=ethereum,
+            method_name='get_gauge',
+            arguments=[pool_address],
+        ))
+        lp_token_address = to_checksum_address(metaregistry.call(
+            node_inquirer=ethereum,
+            method_name='get_lp_token',
+            arguments=[pool_address],
+        ))
+        coins_raw = metaregistry.call(
+            node_inquirer=ethereum,
+            method_name='get_coins',
+            arguments=[pool_address],
+        )
+        coins = []
+        for raw_coin_address in coins_raw:
+            if raw_coin_address == ZERO_ADDRESS:
                 break
-            pool_coins.append(checksumed_coin_addr)
+            coins.append(to_checksum_address(raw_coin_address))
+        underlying_coins_raw = metaregistry.call(
+            node_inquirer=ethereum,
+            method_name='get_underlying_coins',
+            arguments=[pool_address],
+        )
+        underlying_coins = []
+        for raw_underlying_coin_address in underlying_coins_raw:
+            if raw_underlying_coin_address == ZERO_ADDRESS:
+                break
+            underlying_coins.append(to_checksum_address(raw_underlying_coin_address))
 
-        # for metapools pool addr is the lp token as well
-        pools_mapping[pool_addr] = (pool_addr, pool_coins, None)
+        all_pools.append(CurvePoolData(
+            pool_address=pool_address,
+            pool_name=f'{pool_name} curve pool',
```

Comment:
Do you put the suffix ` name` next to every name you write?

```suggestion
            pool_name=f'{pool_name}',
```